prog/dftb+/lib_dftbplus/initprogram.F90

```diff
@@ -1136,11 +1133,15 @@ subroutine initProgramVariables(input, env)
     tSccCalc = input%ctrl%tScc
     tDFTBU = input%ctrl%tDFTBU
     tSpin = input%ctrl%tSpin
-    tREKS = input%ctrl%reksIni%tREKS
-    ! REKS follows spin-restricted open-shell scheme so
-    ! nSpin should be two but some variables such as qOutput
-    ! should be treated in restricted scheme. Here nSpin
-    ! is one and changes to two in the latter part.
+    if (input%ctrl%reksIni%tREKS) then
+      ! REKS follows spin-restricted open-shell scheme so nSpin should be two but some variables
+      ! such as qOutput should be treated in restricted scheme. Here nSpin is one and changes to two
+      ! in the latter part.
+      allocate(reks)
+      if (tSpin) then
+        call error("REKS is not compatible with standard spin polarization")
```

Comment:
This line is repeated check with 2586 line. If you check `tSpin` in this line, then I want to the sentence about spin constants like below (2586 line).

Reply:
As this duplicates the test, I have removed the test here and simplified the logic as well as expanding the comments